R/macro-drat.R

```diff
@@ -0,0 +1,114 @@
+#' do_drat
+#'
+#' The [do_drat()] macro adds the necessary steps for building
+#' and deploying a drat repository to host R package sources.
+#'
+#' @importFrom desc desc_get_version
+#'
+#' @include macro.R
+#' @include macro-package-checks.R
+#' @name macro
+NULL
+
+#' Build and deploy drat repository
+#'
+#' @description
+#' `do_drat()` builds and deploys R packages to a drat repository and adds
+#' default steps to the `"install"`, `"before_deploy"` and `"deploy"` stages:
+#'
+#' @inheritParams step_add_to_drat
+#' @inheritParams step_setup_ssh
+#' @inheritParams step_setup_push_deploy
+#' @inheritParams step_do_push_deploy
+#' @param repo_slug `[string]`\cr
+#'   The name of the drat repository to deploy to in the form `:owner/:repo`.
+#' @param path,branch By default, this macro deploys the `"master"` branch
+#'   of the drat repository. An alternative option is `"gh-pages"`.
+#' @param ssh_key_name `[string]`\cr
+#'   The name of the private SSH key which will be used for deployment.
+#'   Must be available in the build as a secure environment variable.
+#' @param deploy_dev `[logical]`\cr
+#'   Should development versions of packages also be deployed to the drat repo?
+#'   By default only "major", "minor" and "patch" releases are build and
+#'   deployed.
+#'
+#' @section Deployment: Deployment can only happen to the `master` or
+#'   `gh-pages` branch because the Github Pages functionality from Github is
+#'   used to access the drat repository later on. You need to enable this
+#'   functionality when creating the drat repository on Github via `Settings ->
+#'   Github pages` and set it to the chosen setting here.
+#'
+#'   To build and deploy Windows binaries, builds on Travis CI with deployment
+#'   permissions need to be triggered. To build and deploy macOS binaries,
+#'   builds on Travis CI with deployment permissions need to be triggered. Have
+#'   a look at \url{https://docs.ropensci.org/tic/articles/deployment.html} for more
+#'   information and instructions.
+#' @family macros
+#' @export
+#' @examples
+#' \dontrun{
+#' dsl_init()
+#'
+#' do_drat()
+#'
+#' dsl_get()
+#' }
+do_drat <- function(repo_slug = NULL,
+                    orphan = FALSE,
+                    checkout = TRUE,
+                    path = "~/git/drat",
+                    branch = "master",
+                    remote_url = NULL,
+                    commit_message = NULL,
+                    commit_paths = ".",
+                    ssh_key_name = "id_rsa",
```

Comment:
Is this still the correct default?

Reply:
In this case yes. We have to manually insert the private secured key in `appveyor.yml` and I think inserting it with "TRAVIS_DEPLOY_KEY" would be misleading.

Since "id_rsa" is supported universally, I choose to go with it here.
Keys on Appveyor needs to be manually created anyway until now. We could change the name once we have an Appveyor client pkg.

Reply:
I wouldn't mind reverting to `"id_rsa"`, this would save lots of downstream pain.

Reply:
Hm, I know. One reason of the whole effort was to ensure that differently named private SSH key works throughout all functions, especially when installing the key. This came from the motivation to decouple `travis::use_travis_deploy()` from {tic}. I know that this does not mean that the default needs to change necessarily - but having a different one is an indirect way to save the tests whether a custom name would also work 😅 

Also I still think that some non-tech users will have an easier life understanding what that env var is about.

However, I see that we complicate things (a bit) since `id_rsa` is the canonical default picked up everywhere. 

Argh, I don't know what's best 🤔